packages/react-composites/src/composites/CallWithChatComposite/ChatButton/useUnreadMessagesTracker.ts

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { useEffect, useState } from 'react';
+import { ChatAdapter } from '../../ChatComposite/adapter/ChatAdapter';
+import { ChatMessage } from '@azure/communication-chat';
+
+/**
+ * Used by the CallWithChatComposite to track unread messages for showing as a badge on the Chat Button.
+ * @private
+ */
+export const useUnreadMessagesTracker = (chatAdapter: ChatAdapter, isChatPaneVisible: boolean): number => {
+  const [unreadChatMessagesCount, setUnreadChatMessagesCount] = useState<number>(0);
+
+  useEffect(() => {
+    if (isChatPaneVisible) {
+      setUnreadChatMessagesCount(0);
+      return;
+    }
+    const incrementUnreadChatMessagesCount = (event: { message: ChatMessage }): void => {
+      if (!isChatPaneVisible && validNewChatMessage(event.message)) {
+        setUnreadChatMessagesCount(unreadChatMessagesCount + 1);
+      }
+    };
+    chatAdapter.on('messageReceived', incrementUnreadChatMessagesCount);
+
+    return () => {
+      chatAdapter.off('messageReceived', incrementUnreadChatMessagesCount);
+    };
+  }, [chatAdapter, setUnreadChatMessagesCount, isChatPaneVisible, unreadChatMessagesCount]);
+
+  return unreadChatMessagesCount;
+};
+
+/**
+ * Helper function to determine if the message in the event is a valid one from a user.
+ * Display name is used since system messages will not have one.
+ */
+const validNewChatMessage = (message): boolean =>
```

Comment:
nit: Would this break with CTE users that have unnamed participant? (Due to cte bug and no graph integration with calling sdk yet).
cc @PorterNan

Reply:
Great spot! I won't change in this PR as this function was just moved but I think you're right!

Reply:
Filed a bug https://skype.visualstudio.com/SPOOL/_workitems/edit/3342861